packages/bvaughn-architecture-demo/src/hooks/useIndexedDB.ts

```diff
@@ -39,11 +87,18 @@ export default function useIndexedDB<T>(options: {
   const { initialValue, recordName, scheduleUpdatesAsTransitions = false, storeName } = options;
 
   const [isPending, startTransition] = useTransition();
-  const [value, setValue] = useState<T>(initialValue);
+  // If requested, suspend entirely until we've read the initial value
+  //  and can use that to seed the `useState`
+  const [value, setValue] = useState<T>(() => {
+    // In order for persistence to work properly, `getInitialIDBValueAsync` _must_ have been
+    // called in the main app, in `src/ui/setup/index.ts::bootstrapApp()`.
+    return (
+      getInitialIDBValueIfCached(options.database, storeName, recordName)?.value ?? initialValue
+    );
```

Comment:
Should we just lean into this and _throw_ if the value hasn't been preloaded/cached? I kind of think maybe we should. It would simplify the rest of this hook a lot. (We could get rid of the `useEffect` async initialization.)

Reply:
On second thought I'm not sure we _can_ throw here, because we still have to handle the case of no value existing on first access of a given recording, and I'm not sure we can differentiate "wasn't initialized" from "first open".

Reply:
The primary thing I'm concerned about is if we forget to pre-fetch one of these stores, it might behave differently if the IDB value loads faster than other data or not, and that might vary based on environment or recording (which could potentially cause a value to flicker in some cases and not in others).

Reply:
All right, I can try to apply that change.

Reply:
Okay